### DISCUSSION TITLE
Real-time chat using React Native and Pocketbase.

### DISCUSSION THREAD

**Discussion by wesleywcr** [pocketbase/pocketbase#4360] [Show and tell]
Hello guys, I created an application with React Native, Expo, Expo Router and performed some automated tests with Jest, and used PocketBase resources to create my backend. It is a chat application, that has capabilities for sending and receiving messages in real-time, user registration and login, and the possibility of updating registered information. I haven't published my backend but accept host suggestions.

If you want, you can find this project at https://github.com/wesleywcr/chat-fax.

**Comment by jesusmdy** [comment]
Great! how do you handle subscriptions to collections? currently struggling with EventSource when trying to use realtime.

**Reply by ganigeorgiev** [reply]
@jesusmdy This is an old discussion. Please check the [readme of the JS SDK](https://github.com/pocketbase/js-sdk?tab=readme-ov-file#nodejs-via-npm).

In general you need to register `EventSource` polyfill, ex. `react-native-sse`:

```js
import eventsource from "react-native-sse";

global.EventSource = eventsource;
```

Additionally, if you are testing on Android you may have to disable Flipper. See the links in https://github.com/pocketbase/pocketbase/discussions/2484#discussioncomment-6118523.
